File: entites/BlockMap.py
class BlockMap:
    def __init__(self, gridX, gridY, columnNumber, rowNumber, offsets, lineDefIndexByBlock):
        self.gridX = gridX
        self.gridY = gridY
        self.columnNumber = columnNumber
        self.rowNumber = rowNumber
        self.offsets = offsets
        self.lineDefIndexByBlock = lineDefIndexByBlock

    def __str__(self):
        return f"{self.gridX=}, {self.gridY=}, {self.columnNumber=}, {self.rowNumber=}, {self.offsets=}, {self.lineDefIndexByBlock=}"

    __repr__ = __str__

File: cpp_code/BlockMap.cpp
#include <cstdint>
#include <iostream>
#include "./headers/BlockMap.h"

std::ostream& operator<<(std::ostream& os, const BlockMap& obj) {
      os << "BlockMap{ ";
      os << "gridX: " << obj.gridX << " ";
      os << "gridY: " << obj.gridY << " ";
      os << "columnNumber: " << obj.columnNumber << " ";
      os << "rowNumber: " << obj.rowNumber << " ";
      os << "offsets: " << obj.offsets << " ";
      os << "lineDefIndexByBlock: " << obj.lineDefIndexByBlock << " ";
      os << "}";
      return os;
}